front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionSelector.tsx

```diff
@@ -38,37 +56,78 @@ export default function PowerRestrictionSelector() {
     }
   };
 
-  const definePowerRestrictionRange = (powerRestrictionCode?: string) => {
-    if (powerRestrictionCode && pathFindingID) {
-      getPathFindingById({ id: pathFindingID })
-        .unwrap()
-        .then((pathFinding) => {
-          const pathLength = Math.round(
-            lengthFromLineCoordinates(pathFinding?.geographic?.coordinates) * 1000
-          );
-          const powerRestrictionRange: PowerRestrictionRange[] = [
-            {
-              begin_position: 0,
-              end_position: pathLength,
-              power_restriction_code: powerRestrictionCode,
-            },
-          ];
-          dispatch(updatePowerRestriction(powerRestrictionRange));
-        });
+  // Extract uniques rollingstock's power restriction codes allowed by each electrification mode
+  const cleanConditionalEffortCurves = (rollingStockToClean: RollingStock) => {
+    const curvesMode = rollingStockToClean.effort_curves.modes;
+    const curvesModesKey = Object.keys(curvesMode);
+
+    let parsedElectrification: ElectrificationPR = {};
+    for (const mode of curvesModesKey) {
+      let powerCodes = curvesMode[mode].curves.map((curve) => curve.cond?.power_restriction_code);
+      powerCodes = powerCodes.filter(
+        (value, index, array) => value && array.indexOf(value) === index
+      );
+      parsedElectrification = {
+        ...parsedElectrification,
+        [mode]: powerCodes,
+      };
+    }
+    return parsedElectrification;
+  };
+
+  const definePowerRestrictionRange = async (powerRestrictionCode?: string) => {
+    if (powerRestrictionCode && pathFinding) {
+      const pathLength = Math.round(
+        lengthFromLineCoordinates(pathFinding.geographic?.coordinates) * 1000
+      );
+      const powerRestrictionRange: PowerRestrictionRange[] = [
+        {
+          begin_position: 0,
+          end_position: pathLength,
+          power_restriction_code: powerRestrictionCode,
+        },
+      ];
+      dispatch(updatePowerRestriction(powerRestrictionRange));
+
+      if (rollingStock && pathWithCatenaries) {
+        const parsedElectrification = cleanConditionalEffortCurves(rollingStock);
+
+        const pathCatenaryRanges = pathWithCatenaries.catenary_ranges;
+        if (pathCatenaryRanges) {
+          // Extract path electrification mode and check compatibility
+          const pathElectrification = pathCatenaryRanges.map((range) => range.mode);
```

Comment:
```suggestion
          const pathElectrification = compact(pathCatenaryRanges.map((range) => range.mode));
```
Add a compact here to remove the null and undefined values

Reply:
Done